src/types.h

```diff
@@ -288,13 +288,18 @@ inline T& operator--(T& d) { return d = T(int(d) - 1); }
 ENABLE_BASE_OPERATORS_ON(T)                                        \
 constexpr T operator*(int i, T d) { return T(i * int(d)); }        \
 constexpr T operator*(T d, int i) { return T(int(d) * i); }        \
+constexpr T operator&(int d1, T d2) { return T(int(d1) & int(d2));} \
+inline T& operator&=(T& d1, T d2) { return d1 = d1 & d2; }          \
+constexpr T operator|(int d1, T d2) { return T(int(d1) | int(d2));} \
+inline T& operator|=(T& d1, T d2) { return d1 = d1 | d2; }          \
```

Comment:
Perhaps `Value` deserves a special macro `ENABLE_MATH_OPERATORS_ON(T)` which does not include bit-wise operations.

Reply:
Since they are templates, If we don't ever request Value |= Value, then will an actual operator be implemented?

Reply:
I made a BITWISE operators section and added that to separate base math from bitwise.  I think that's a bit cleaner.